scripts/generate_image.ps1

```diff
@@ -0,0 +1,175 @@
+# License is located at
+#
+#	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# This script is used to generate windows docker images used by agent
+# integration tests and functions tests and upload images to ECR.
+
+Param(
+    [string]$GIT_REPO,
+    [string]$COMMIT_SHA,
+    [string]$S_REPOSITORY,
+    [string]$R_REPOSITORY
+)
+
+$ErrorActionPreference = 'SilentlyContinue'
+
+# Constants
+$S_REGION="us-west-2"
+$R_REGION="cn-north-1"
+$GitVersion="2.10.1"
+$GoVersion="1.9"
+$WORK_ROOT="${env:TMP}\amazon-ecs-agent"
+$GOPATH="${WORK_ROOT}\go"
+$BUILD_ROOT="${GOPATH}\src\github.com\aws\amazon-ecs-agent"
+
+if(!$(Test-Path -Path "${WORK_ROOT}")) {
+  New-Item -Path "${WORK_ROOT}" -Type directory
+}
+
+echo "==============================================================================="
+echo "Building windows images for commit ${COMMIT_SHA}"
+echo "==============================================================================="
+
+# Install Git
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+echo "Downloading Git..."
+Invoke-WebRequest "https://github.com/git-for-windows/git/releases/download/v${GitVersion}.windows.1/Git-${GitVersion}-64-bit.exe" -OutFile "${env:TEMP}\git-installer.exe" -UseBasicParsing
+echo "Installing Git..."
+Start-Process $env:TEMP\git-installer.exe -ArgumentList '/SILENT /COMPONENTS="icons,ext,ext\reg\shellhere,assoc,assoc_sh"' -NoNewWindow -Wait
+$env:Path = ${env:Path} + ";${env:ProgramFiles}/Git/cmd"
+[Environment]::SetEnvironmentVariable("Path", $env:Path + ";${env:ProgramFiles}/Git/cmd", [EnvironmentVariableTarget]::Machine)
+git config --global user.name "OctoKitten"
+git config --global user.email "nobody@example.localhost"
+
+# Install Go
+echo "Downloading Go..."
+Invoke-WebRequest "https://storage.googleapis.com/golang/go${GoVersion}.windows-amd64.msi" -OutFile "${env:TEMP}\go-installer.msi" -UseBasicParsing
+echo "Installing Go..."
+Start-Process "msiexec.exe" -ArgumentList "/quiet /i ${env:TEMP}\go-installer.msi" -Wait
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\bin"
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\src"
+[Environment]::SetEnvironmentVariable("Path", $env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin", [EnvironmentVariableTarget]::Machine)
+[Environment]::SetEnvironmentVariable("GOPATH", "${WORK_ROOT}\go", [EnvironmentVariableTarget]::Machine)
+$env:Path=$env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin"
+$env:GOPATH="${WORK_ROOT}\go"
+
+echo "Checking out ${GIT_REPO} at ${COMMIT_SHA}"
+git clone "${GIT_REPO}" "${BUILD_ROOT}" 2>&1 | % {$_.ToString()}
+cd "${BUILD_ROOT}"
+git config --add remote.origin.fetch '+refs/pull/*/head:refs/remotes/origin/pr/*'
+git fetch 2>&1 | % {$_.ToString()}
```

Comment:
You mean " 2>&1 | % {$_.ToString()}"? Git is printing some weird thing to stderr. Like 
It prints the checking out files progress to stderr instead of stdout and also something like this:

"You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout."

I have logic to check stderr in my script to check the status of running. So I need to redirect it to stdout. The % {$_.ToString()} is just to format the output.